x-pack/test/functional/apps/lens/smokescreen.ts

```diff
@@ -0,0 +1,110 @@
+export default function({ getService, getPageObjects }: FtrProviderContext) {
+  const PageObjects = getPageObjects([
+    'header',
+    'common',
+    'visualize',
+    'dashboard',
+    'header',
+    'timePicker',
+    'lens',
+  ]);
+  const find = getService('find');
+  const dashboardAddPanel = getService('dashboardAddPanel');
+
+  async function assertExpectedMetric() {
+    await PageObjects.lens.assertExactText(
+      '[data-test-subj="lns_metric_title"]',
+      'Maximum of bytes'
+    );
+    await PageObjects.lens.assertExactText('[data-test-subj="lns_metric_value"]', '19,986');
+  }
+
+  async function assertExpectedTable() {
+    await PageObjects.lens.assertExactText(
+      '[data-test-subj="lnsDataTable"] thead .euiTableCellContent__text',
+      'Maximum of bytes'
+    );
+    await PageObjects.lens.assertExactText(
+      '[data-test-subj="lnsDataTable"] tbody .euiTableCellContent__text',
+      '19,986'
+    );
+  }
+
+  describe('lens smokescreen tests', () => {
+    it('should allow editing saved visualizations', async () => {
+      await PageObjects.visualize.gotoVisualizationLandingPage();
+      await PageObjects.lens.clickVisualizeListItemTitle('Artistpreviouslyknownaslens');
+      await PageObjects.lens.goToTimeRange();
+      await assertExpectedMetric();
+    });
+
+    it('should be embeddable in dashboards', async () => {
+      await PageObjects.common.navigateToApp('dashboard');
+      await PageObjects.dashboard.clickNewDashboard();
+      await dashboardAddPanel.clickOpenAddPanel();
+      await find.clickByButtonText('Artistpreviouslyknownaslens');
+      await dashboardAddPanel.closeAddPanel();
+      await PageObjects.lens.goToTimeRange();
+      await assertExpectedMetric();
+    });
+
+    it('should allow seamless transition to and from table view', async () => {
+      await PageObjects.visualize.gotoVisualizationLandingPage();
+      await PageObjects.lens.clickVisualizeListItemTitle('Artistpreviouslyknownaslens');
+      await PageObjects.lens.goToTimeRange();
+      await assertExpectedMetric();
+      await PageObjects.lens.switchToVisualization('lnsChartSwitchPopover_lnsDatatable');
+      await assertExpectedTable();
+      await PageObjects.lens.switchToVisualization('lnsChartSwitchPopover_lnsMetric');
+      await assertExpectedMetric();
+    });
+
+    it('should allow creation of lens visualizations', async () => {
+      await PageObjects.visualize.navigateToNewVisualization();
+      await PageObjects.visualize.clickVisType('lens');
+      await PageObjects.lens.toggleExistenceFilter();
+      await PageObjects.lens.goToTimeRange();
+
+      // Drag the @timestamp field to the workspace
+      await PageObjects.lens.dragToWorkspace('@timestamp');
+
+      // Change the y from count to average of bytes
+      await PageObjects.lens.changeDimension({
+        from: 'Count of documents',
+        to: 'avg',
+        field: 'bytes',
+      });
```

Comment:
When I was running the tests in the background, this await hung forever. It looks like it goes into an infinite loop looking for the text `Count of documents` but the drag never finished from the previous line.

Reply:
Yeah. That's a drag and drop behavioral issue I'm still working through. Currently drag and drop only works if the actual mouse is hovered over the correct drop target! I may just change this to not test drag and drop for now, and solve that later. I've got one last stab at it currently running in CI.